tests/udf/test_pick.py

```diff
@@ -44,3 +44,47 @@ def test_pick_empty_roi(lt_ctx, delayed_ctx):
 
     assert data.dtype == res['intensity'].data.dtype
     assert data.dtype == res_delayed['intensity'].data.dtype
+
+
+def test_pick_shifted(lt_ctx, delayed_ctx):
+    data = _mk_random(size=(16, 16, 16, 16), dtype="float32")
+    # data = np.ones((16, 16, 16, 16), dtype="float32")
+    # data = np.arange(0, 16*16*16*16, dtype="float32").reshape((16, 16, 16, 16))
+    dataset = MemoryDataSet(data=data, tileshape=(3, 7, 16),
+                            num_partitions=7, sig_dims=2)
+    roi = np.random.choice([True, False], size=dataset.shape.nav)
+    roi[0] = True
+
+    regression_coefficients = np.array([(3, 3), (0, 0), (0, 0)])
+
+    udf = PickShiftedUDF(regression_coefficients=regression_coefficients)
+    res = lt_ctx.run_udf(dataset=dataset, udf=udf, roi=roi)
+    res_delayed = delayed_ctx.run_udf(dataset=dataset, udf=udf, roi=roi)
+
+    assert np.allclose(data[roi][..., 3:, 3:], res['intensity'].data[..., :-3, :-3], 1e-15)
+    assert np.allclose(data[roi][..., 3:, 3:], res_delayed['intensity'].data[..., :-3, :-3], 1e-15)
+
+    assert data.dtype == res['intensity'].data.dtype
+    assert data.dtype == res_delayed['intensity'].data.dtype
+
+
+def test_pick_shifted_empty_roi(lt_ctx, delayed_ctx):
+    data = _mk_random(size=(16, 16, 16, 16), dtype="float32")
+    dataset = MemoryDataSet(data=data, tileshape=(3, 7, 7),
+                            num_partitions=7, sig_dims=2)
+    roi = np.zeros(dataset.shape.nav, dtype=bool)
+
+    regression_coefficients = np.array([(3, 3), (0, 0), (0, 0)])
+
+    udf = PickShiftedUDF(regression_coefficients=regression_coefficients)
+    res = lt_ctx.run_udf(dataset=dataset, udf=udf, roi=roi)
+    res_delayed = delayed_ctx.run_udf(dataset=dataset, udf=udf, roi=roi)
+
+    assert np.allclose(data[roi], res['intensity'].data)
+    assert np.allclose(data[roi], res_delayed['intensity'].data)
+
+    assert data[roi].shape == res['intensity'].data.shape
+    assert data[roi].shape == res_delayed['intensity'].data.shape
+
+    assert data.dtype == res['intensity'].data.dtype
+    assert data.dtype == res_delayed['intensity'].data.dtype
```

Comment:
The regression coefficients should all be different to make sure we are not mixing them anywhere. In particular, the dy/dx ... coefficients require tests.

We should also confirm that the coefficients act the same as in `CoMUDF`, in particular if we support more parameters, as suggested above. A simple and comprehensive test would be comparing `CoMUDF` on an uncorrected dataset vs. `CoMUDF` on a corrected dataset.